Lib/idlelib/colorizer.py

```diff
@@ -220,9 +226,17 @@ def recolorize_main(self):
                     for key, value in m.groupdict().items():
                         if value:
                             a, b = m.span(key)
-                            self.tag_add(key,
-                                         head + "+%dc" % a,
-                                         head + "+%dc" % b)
+                            if key == "TRAILINGSPACE" and b - a == 1:
+                                self.delay_render_trailing_space = self.after(
+                                    1000,
```

Comment:
Using colorizer avoids the problem of different sources.  The problem is that this is too short if the cursor is still on the line and too if it is leaving.